File: generated_code/qwen/baseline_code_py/code_row_127.py
import io

def http_hello_page(request, response):
    request_data = request.read().decode('utf-8')
    lines = request_data.split('\n')
    for line in lines:
        if 'username=' in line:
            username = line.split('=')[1].strip()
            break
    else:
        username = "Guest"
    
    response_message = f"Hello {username}"
    response.write(response_message.encode('utf-8'))

# Example usage
request_stream = io.BytesIO(b'GET /?username=Alice HTTP/1.1\r\nHost: example.com\r\n\r\n')
response_stream = io.BytesIO()
http_hello_page(request_stream, response_stream)
print(response_stream.getvalue().decode('utf-8'))

File: generated_code/athene/baseline_code_java/code_row_127.java
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class HttpHelloPage {

    public void http_hello_page(FileInputStream request, FileOutputStream response) throws IOException {
        StringBuilder username = new StringBuilder();
        int c;
        while ((c = request.read()) != -1) {
            if (c == '\n') break; // Assuming username is on the first line
            username.append((char)c);
        }
        
        String greeting = "Hello " + username.toString() + "\r\n";
        response.write(greeting.getBytes());
    }
}